lib/PuppeteerSharp/Messaging/MessageResponse.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Newtonsoft.Json;
+
+namespace PuppeteerSharp.Messaging
+{
+    internal class MessageResponse
+    {
+        [JsonProperty("requestId")]
+        public string RequestId { get; set; }
+
+        [JsonProperty("request")]
+        public Request Request { get; set; }
```

Comment:
This is looking good. Maybe it seems too much, but the final code looks better.
I wouldn't reuse the Request and Response classes though. One thing is data and another the complex behavior of those classes.
I would buy a ResposeData and RequestData class and make the Response/Request class inherit from them.